packages/components/src/components/popover/popover.tsx

```diff
@@ -29,51 +29,69 @@ const Popover = ({
     relative_render = false,
     should_disable_pointer_events = false,
     should_show_cursor,
-    window_border,
     zIndex = '1',
     data_testid,
     arrow_styles,
 }: React.PropsWithChildren<TPopoverProps>) => {
     const ref = React.useRef<HTMLDivElement | undefined>();
     const [popover_ref, setPopoverRef] = React.useState<HTMLDivElement | undefined>(undefined);
-
+    const [is_bubble_visible, setIsBubbleVisible] = React.useState(false);
+    const { is_mobile } = useDevice();
     const [hover_ref, is_hovered] = useHover(null, true);
     const [bubble_hover_ref, is_bubble_hovered] = useHoverCallback();
+    const should_toggle_on_target_tap = React.useMemo(() => is_mobile && is_open === undefined, [is_mobile, is_open]);
 
     React.useEffect(() => {
         if (ref.current) {
             setPopoverRef(ref.current);
         }
     }, [has_error]);
+    React.useEffect(() => {
+        if (!is_hovered && should_toggle_on_target_tap) {
+            setIsBubbleVisible(false);
+        }
+    }, [is_hovered, should_toggle_on_target_tap]);
 
     const onMouseEnter = () => {
-        if (onBubbleOpen) onBubbleOpen();
+        if (onBubbleOpen) {
+            if (should_toggle_on_target_tap) setIsBubbleVisible(true);
```

Comment:
Should we check `should_toggle_on_target_tap` here if it is set to true only for mobile?

Reply:
`onMouseEnter` will be firing both in Chrome desktop & responsive, and I don't want to set `is_bubble_visible` to `true` here when `onMouseEnter` fires while `onBubbleOpen` is passed in props, if it's desktop or if popover is controllable, e.g. 'Token is copied!' popover.
I want to set `is_bubble_visible` to true here when both conditions are true:
- it's mobile/responsive & popover is uncontrollable
- `onMouseEnter` fires & `onBubbleOpen` is passed.